docs/source/app-dev/services.rst

```diff
@@ -56,9 +56,12 @@ A call to the command submission service will return as soon as the ledger serve
 
 The on-ledger effect of the command execution will be reported via the `transaction service <#transaction-service>`__, described below. The completion status of the command is reported via the `command completion service <#command-completion-service>`__. Your application should receive completions, correlate them with command submission, and handle errors and failed commands. Alternatively, you can use the `command service <#command-service>`__, which conveniently wraps the command submission and completion services.
 
-Commands can be labeled with two application-specific IDs, both of which are returned in completion events:
+Each intended ledger change is identified by the **change ID**: the triplet of the submitting :ref:`party <com.daml.ledger.api.v1.Commands.party>`, :ref:`application ID <com.daml.ledger.api.v1.Commands.application_id>` and :ref:`command ID <com.daml.ledger.api.v1.Commands.command_id>`.
+
+Commands can be labeled with application-specific IDs, all of which are returned in completion events:
 
 - A :ref:`commandId <com.daml.ledger.api.v1.Commands.command_id>`, returned to the submitting application only. It is generally used to implement this correlation between commands and completions. 
```

Comment:
```suggestion
- A :ref:`commandId <com.daml.ledger.api.v1.Commands.command_id>`, returned to the submitting application only; it can be used to correlate commands to completions.
```
I'd perhaps move it after the submission ID, which seems more important to me.

Reply:
8b7eec5